src/doctor.ts

```diff
@@ -0,0 +1,66 @@
+import { getProtocolInterface } from "./deps.ts";
+
+type RuntimeVersion = {
+  name: string;
+  current: string;
+  minimum?: string;
+  error?: {
+    message: string;
+  };
+};
+
+const getHostedDenoRuntimeVersion = async (): Promise<{
+  minimum?: string;
+  message?: string;
+  error?: { message: string };
+}> => {
+  try {
+    const metadataURL = "https://api.slack.com/slackcli/metadata.json";
+    const response = await fetch(metadataURL);
+    if (!response.ok) {
+      throw new Error("Failed to collect upstream CLI metadata");
+    }
+    const metadata = await response.json();
+    const version = metadata?.["deno-runtime"]?.releases[0]?.version;
+    if (!version) {
+      throw new Error("Failed to find the minimum Deno version");
+    }
+    const message = Deno.version.deno !== version
+      ? `Applications deployed to Slack use Deno version ${version}`
+      : undefined;
+    return { minimum: version, message };
```

Comment:
Yeah we discussed this point in the spec @zimeg wrote. The `minimum` is just the payload shape, but you can see in the messaging to the user we don't actually mention minimum. This is intentional. We are very carefully choosing our words when raising this to the user so that we don't imply a 'minimum' version but rather communicate that ROSI uses one EXACT version.

Reply:
I agree that "minimum" isn't a super ideal word for this case but it gives us flexibility across other SDKs to specify certain runtime requirements IMO.

With the `current` and `minimum` version comparison moving into the hooks I'm super interested in how this output might change. Or maybe it doesn't? But wanted to share an example for reference:

![minimum-error](https://github.com/slackapi/deno-slack-hooks/assets/18134219/24a5afef-174b-4bf8-bcf1-a2988ae8ba15)

Reply:
Without special error handling in the CLI, the hook can output this messaging which I think is pretty alright! All of this info is coming from the hook.

<img width="626" alt="unsupported-runtime" src="https://github.com/slackapi/deno-slack-hooks/assets/18134219/40bd0f3b-007b-4c53-95f2-efb1906648ee">

Reply:
LGTM

Reply:
Just to circle back on this discussion with the latest updates- `minimum` no longer exists. Neither does `latest`. At least in the hook response. Only `message` and `errors.message` can provide additional detail.

Reply:
LGTM also 👍 
The hook can also use the protocol to share an error message correct?